triton-client/src/components/DatasetColumns.tsx

```diff
@@ -0,0 +1,371 @@
+    const totalSize = dataset.totalSize
+    const datasetID = dataset.id
+
+    const dispatch = useAppDispatch()
+    const [updatingRequest, setUpdatingRequest] = useState(false)
+
+    const activeRequest = dataset.activeRequest
+    const alreadyRequested = !!activeRequest
+    const currentRequest =
+        activeRequest?.type === type ? activeRequest : undefined
+
+    const project = useAppSelector((state) =>
+        dataset?.external_project_id
+            ? state.projectsState.projectsById[dataset.external_project_id]
+            : undefined,
+    )
+    const constants = useAppSelector(selectConstants)
+
+    const dispatchCreateRequest = useCallback(
+        async (type: DownloadRequestType) => {
+            if (dataset) {
+                setUpdatingRequest(true)
+                await dispatch(
+                    createDownloadRequest(
+                        dataset.external_project_id,
+                        dataset.id,
+                        type,
+                    ),
+                )
+                    .then(() => {
+                        updateDataset(datasetID)
+                    })
+                    .finally(() => {
+                        setUpdatingRequest(false)
+                    })
+            }
+        },
+        [dataset, dispatch],
+    )
+
+    const request = useCallback(
+        (downloadType: DownloadRequestType) => {
+            if (dataset && project && totalSize) {
+                const diskUsage = project.diskUsage[downloadType]
+                const diskCapacity = constants.diskCapacity[downloadType]
+                if (diskUsage + totalSize > diskCapacity) {
+                    Modal.confirm({
+                        title: `${downloadType} Project Quota Exceeded`,
+                        content: `The total size of the datasets will exceed the ${downloadType} project quota. This dataset will be queued until space is freed.`,
+                        onOk: () =>
+                            dispatchCreateRequest(downloadType).catch((e) =>
+                                console.error(e),
+                            ),
+                        okText: "Continue",
+                        cancelText: "Cancel",
+                    })
+                } else {
+                    dispatchCreateRequest(downloadType).catch((e) =>
+                        console.error(e),
+                    )
+                }
+            }
+        },
+        [
+            constants.diskCapacity,
+            dataset,
+            dispatchCreateRequest,
+            project,
+            totalSize,
+        ],
+    )
+
+    if (
+        activeRequest !== undefined &&
+        currentRequest == activeRequest &&
+        !currentRequest.should_delete &&
+        currentRequest.status !== "FAILED"
+    ) {
+        const { type, status } = currentRequest
+        const actions: ActionDropdownProps["actions"] = [
```

Comment:
this dropdown can be simplified in a similar following manner : 
`        await resetPassword(projectExternalId, type).then(
            () =>
                notification.success({
                    message: "Password Reset",
                    description: `The ${type} password is scheduled for reset for the project ${project.external_name}.`,
                }),
            (reason) => {
                notification.error({
                    message: "Error",
                    description: `The ${type} password could not be reset for the project ${project.external_name}. ${reason}`,
                })
            },
        )
    }`